pyuvdata/calfits.py

```diff
@@ -273,8 +285,13 @@ def write_calfits(self, filename, spoof_nonessential=False,
             hdulist = fits.HDUList([prihdu, ant_hdu, sechdu])
 
         else:
-            prihdu = fits.PrimaryHDU(data=pridata, header=prihdr)
-            hdulist = fits.HDUList([prihdu, ant_hdu])
+            if self.total_quality_array is not None:
+                prihdu = fits.PrimaryHDU(data=pridata, header=prihdr)
+                sechdu = fits.ImageHDU(data=secdata, header=sechdr)
+                hdulist = fits.HDUList([prihdu, ant_hdu, sechdu])
+            else:
+                prihdu = fits.PrimaryHDU(data=pridata, header=prihdr)
+                hdulist = fits.HDUList([prihdu, ant_hdu])
 
```

Comment:
I hadn't done it since there already was a secondary HDU in the delay type. For backwards compatibility, I've added the total_quality_array as a tertiary HDU for delay cals, rather than make it the secondary for consistency between types. Is this solution acceptable?